### ISSUE TITLE
urllib3 PoolManger issue on pypy3 tests

### ISSUE THREAD

**Issue by shalevr** [open-telemetry/opentelemetry-python-contrib#1827]
There is an issue with the new urllib3 version(1.26.16) on pypy3, the [test_basic_http_success_using_connection_pool](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/a6797541721af72e6feab400e24ceaabc08cb31e/instrumentation/opentelemetry-instrumentation-urllib3/tests/test_urllib3_ip_support.py#L51) test doesn't finish.

According to the urllib3 documentation the new version fixed:

Fixed thread-safety issue where accessing a ``PoolManager`` with many distinct origins
  would cause connection pools to be closed while requests are in progress (`#2954 <https://github.com/urllib3/urllib3/pull/2954>`_)

It might be related to the [HttpTestbase](https://github.com/open-telemetry/opentelemetry-python/blob/46757ff063d7f6657e7eeef17350f08aa0f1e42f/tests/opentelemetry-test-utils/src/opentelemetry/test/httptest.py#LL22C1-L22C1) and to the HTTPServer thread.

**Comment by shalevr** [comment]
In the meantime, I opened PR that skipped the urlib3 tests on pypy3
#1826

**Comment by srikanthccv** [comment]
Yeah, I think this is an issue with how we test because both urllib3 and requests have pypy tests as part of their CI workflows, so if it works there, it should pass here.

**Comment by shalevr** [comment]
> Yeah, I think this is an issue with how we test because both urllib3 and requests have pypy tests as part of their CI workflows, so if it works there, it should pass here.

Exactly, it's probably the HttpTestBase class that both urllib3 and requests inherit from it.
